src/include/columnar/columnar_storage.h

```diff
@@ -0,0 +1,54 @@
+/*-------------------------------------------------------------------------
+ *
+ * columnar_storage.h
+ *
+ * Type and function declarations for storage of columnar data in blocks.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#ifndef COLUMNAR_STORAGE_H
+#define COLUMNAR_STORAGE_H
+
+#include "postgres.h"
+
+#include "storage/smgr.h"
+#include "utils/rel.h"
+
+
+/*
+ * Logical offsets never fall on the first two physical pages. See
+ * comments in columnar_storage.c.
+ */
+#define ColumnarInvalidLogicalOffset 0
+#define ColumnarFirstLogicalOffset ((BLCKSZ - SizeOfPageHeaderData) * 2)
+#define ColumnarLogicalOffsetIsValid(X) ((X) >= ColumnarFirstLogicalOffset)
+
+
+extern void ColumnarStorageInit(SMgrRelation srel, uint64 storageId);
+extern bool ColumnarStorageIsCurrent(Relation rel);
+extern void ColumnarStorageUpdateCurrent(Relation rel, bool upgrade,
+										 uint64 reservedStripeId,
+										 uint64 reservedRowNumber,
+										 uint64 reservedOffset);
+
+extern uint64 ColumnarStorageGetVersionMajor(Relation rel, bool force);
+extern uint64 ColumnarStorageGetVersionMinor(Relation rel, bool force);
+extern uint64 ColumnarStorageGetStorageId(Relation rel, bool force);
+extern uint64 ColumnarStorageGetReservedStripeId(Relation rel, bool force);
+extern uint64 ColumnarStorageGetReservedRowNumber(Relation rel, bool force);
+extern uint64 ColumnarStorageGetReservedOffset(Relation rel, bool force);
```

Comment:
maybe `force` -> `forceVersionConsistency` or something

Reply:
As discussed, we will leave this as-is. The word `force` (kind of like `-f`) communicates to the caller that they should be careful and use it in controlled situations like upgrade.